src/vl-map-action.js

```diff
@@ -80,30 +117,38 @@ export class VlMapAction extends vlElement(HTMLElement) {
    */
   actionChanged() {
     const event = new Event(VlMapAction.NEW_ACTION_EVENT_NAME);
-    this.parentElement.dispatchEvent(event);
+    this._mapElement.dispatchEvent(event);
   }
 
   _layerChangedCallback() {
-    this._computeAction(this._map, this.layer);
+    this._mapElement.ready.then(() => {
+      if (this.layer) {
+        const action = this._createAction(this.layer);
+        if (action) {
+          this._mapElement.addAction(action);
+          this.actionChanged();
```

Comment:
da zijn 2 calls na mekaar op hetzelfde element (this._mapElement).
dat doet mij vermoeden dat dat eigenlijk in mapElement moet zitten?

Reply:
Inderdaad, refactoring uitgevoerd.